optimum/neuron/pipelines/transformers/base.py

```diff
@@ -123,7 +123,7 @@ def load_pipeline(
             model, export=export, **compiler_args, **input_shapes, **hub_kwargs, **kwargs
         )
     # uses neuron model
-    elif isinstance(model, NeuronBaseModel):
+    elif isinstance(model, (NeuronBaseModel, NeuronModelForCausalLM)):
```

Comment:
QQ: why is `NeuronModelForCausalLM` not a sublass of `NeuronBseModel`?

Reply:
Because `NeuronBaseModel` is based on JIT models and implements the corresponding conversion logic. `NeuronModelForCausalLM` is a subclass of  `NeuronDecoderModel` that uses `transformers-neuronx` models instead.
We could refactor to add a common class to both though, since the latest subclasses of `NeuronBaseModel` are overriding pretty much all its methods now.

Reply:
Wouldn't it make more sense to use `NeuronDecoderModel` here?

Reply:
I guess it would be good to refactor it if possible. But also it's not pressing and can be postponed to when we have more bandwidth!

Reply:
I will wait until @JingyaHuang 's pull-request on T5 is merged as it is a new subclass, and any change I make before that to the base class will result in nightmarish conflicts.